tc/core/tensor-inl.h

```diff
@@ -63,6 +63,25 @@ void DLTensorDeleter::operator()(const DLTensorType* t) {
   delete t;
 };
 
+template <typename DLTensorType>
+std::string printSizesAndStrides(const DLTensorType* t) {
+  std::ostringstream sizesandshapes;
+  sizesandshapes << "Dimensions of the tensor are: ";
```

Comment:
This is probably too much spew. 
How about `os << "sizes: "` ... `os << " strides: "` ... etc
Would be good to also print the base pointer, the offset and probably the type too